### PR TITLE
Clarify ConfigureAppConfiguration

### PR DESCRIPTION
**Pull request by galakt**
Related to https://github.com/dotnet/runtime/pull/44254

### CODE REVIEW

File: xml/Microsoft.Extensions.Hosting/HostBuilder.xml [hunk 1]
```diff
@@ -104,9 +104,11 @@
       <Docs>
         <param name="configureDelegate">The delegate for configuring the <see cref="T:Microsoft.Extensions.Configuration.IConfigurationBuilder" /> that will be used
             to construct the <see cref="T:Microsoft.Extensions.Configuration.IConfiguration" /> for the application.</param>
-        <summary>Sets up the configuration for the remainder of the build process and application. This can be called multiple times and
-            the results will be additive. The results will be available at <see cref="P:Microsoft.Extensions.Hosting.HostBuilderContext.Configuration" /> for
-            subsequent operations, as well as in <see cref="P:Microsoft.Extensions.Hosting.IHost.Services" />.</summary>
+        <summary>Sets up the configuration for the remainder of the build process and application.
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.Hosting/HostBuilder.xml:107]: What is a `DI container`? Any chance you can expand that acronym?

Comment (maryamariyan) [reply xml/Microsoft.Extensions.Hosting/HostBuilder.xml:107]: DI, dependency injection container, perhaps it's ok to keep `DI container` as-is. there are other instances of uses for DI in docs:

https://github.com/dotnet/dotnet-api-docs/search?q=di+container

File: xml/Microsoft.Extensions.Hosting/HostBuilder.xml [hunk 2]
```diff
@@ -104,9 +104,11 @@
...
+            The <see cref="P:Microsoft.Extensions.Hosting.HostBuilderContext.Configuration"/> passed in is the host's configuration built
+            from calls to <see cref="M:Microsoft.Extensions.Hosting.IHostBuilder.ConfigureHostConfiguration(System.Action{Microsoft.Extensions.Configuration.IConfigurationBuilder})"/>. This can be called multiple times and the results will be additive.
+            After all calls have been processed, <see cref="P:Microsoft.Extensions.Hosting.HostBuilderContext.Configuration"/> will be updated with the results
+            for future build steps. The resulting configuration will also be available in the <see cref="P:Microsoft.Extensions.Hosting.IHost.Services"/> DI Container.</summary>
```
Comment (maryamariyan) [inline xml/Microsoft.Extensions.Hosting/HostBuilder.xml:111]: @carlossanlop is it alright to break summary into multiple lines? just checking.

**Review by maryamariyan — APPROVED** [review]

File: xml/Microsoft.Extensions.Hosting/HostBuilder.xml (hunk 2 above)
Comment (carlossanlop) [reply xml/Microsoft.Extensions.Hosting/HostBuilder.xml:111]: While it's not forbidden, you have to keep in mind that this information is going to show up in Visual Studio via Intellisense when invoking this API. So if the text takes too many lines, the dialog may end up being too large.

When the text is too large for a _summary_, I normally suggest to split the text and put the less important information inside the remarks section.

@maryamariyan up to you to accept it as is, or suggest to put it all in one line, or to move some of the text into the remarks.

**Comment by carlossanlop** [comment]
@galakt thanks for making the change. To answer the question you left in https://github.com/dotnet/runtime/pull/44254 : We will soon bring back these comments to runtime as triple slash, and will substitute everything there. But we also generate intellisense periodically, which reaches developers when invoking these APIs through Visual Studio. So it's good to have the change done here first.

**Comment by galakt** [comment]
@maryamariyan @eerhardt ping

**Pull request merged**